lib/UserAgentApplication.ts

```diff
@@ -183,33 +183,39 @@ namespace Msal {
         * @param _tokenReceivedCallback -  The function that will get the call back once this API is completed (either successfully or with a failure).
         * @param {boolean} validateAuthority -  boolean to turn authority validation on/off.
         */
-        constructor(clientId: string, authority: string, tokenReceivedCallback: tokenReceivedCallback, validateAuthority?: boolean) {
+        constructor(
+            clientId: string,
+            {
+                tokenReceivedCallback,
+                validateAuthority,
+                cacheLocation = 'sessionStorage',
+                authority = 'https://login.microsoftonline.com/common',
+            }: {
+                tokenReceivedCallback?: tokenReceivedCallback,
```

Comment:
@austindyoung tokenReceivedCallback is not an optional parameter. We can put all other parameters after tokenReceivedCallback in a destructured manner instead. I need to add other optional parameters as well so I will submit a pull request for this. Thank you for pointing out this pattern.